pkg/controllers/management/clusterprovisioner/provisioner.go

```diff
@@ -973,20 +979,29 @@ func GetBackupFilename(backup *v3.EtcdBackup) string {
 	return snapshot
 }
 
-// transform an imported cluster into a k3s cluster using its discovered version
-func k3sClusterConfig(cluster *v3.Cluster) error {
+// transform an imported cluster into a k3s or k3os cluster using its discovered version
+func k3sClusterConfig(cluster *v3.Cluster, nodes []*v3.Node) error {
 	// version is not found until cluster is provisioned
-	if cluster.Status.Driver == "" || cluster.Status.Version == nil {
+	if cluster.Status.Driver == "" || cluster.Status.Version == nil || len(nodes) == 0 {
```

Comment:
Is it possible for `nodes == 0` check to block the normal `k3s` cluster from coming through here ? I'm not familiar here, but assuming k3s would still register 1 node even if it has no workers, and then it'd be ok.

Reply:
Might slow down provisioning but I don't think it would block